### PR TITLE
kernel: Unify the pid hash

### PR DESCRIPTION
**Pull request by jeongchanKim**
In each module, there is a pid hash definition, but all are same. So integrated them to one.

### CODE REVIEW

File: os/include/tinyara/sched.h [hunk 1]
```diff
@@ -183,6 +183,9 @@
 #define KEY_NOT_INUSE   (0)
 #define KEY_INUSE       (1)
 
+#define MAX_PID_MASK   (CONFIG_MAX_TASKS - 1)
+#define PIDHASH(pid)  ((pid) & MAX_PID_MASK)
```
Comment (sunghan-chang) [inline os/include/tinyara/sched.h]: Could you align them?

File: external/libtuv/source/tinyara/uv_tinyara_signal.c [hunk 2]
```diff
@@ -131,7 +129,7 @@ static uv_signal_t* uv__signal_first_handle(int signum) {
   lookup.signum = signum;
   lookup.loop = NULL;
 
-  index = PID_HASH(getpid());
+  index = PIDHASH(getpid());
   if (index < 0 || index >= CONFIG_MAX_TASKS) {
```
Comment (YiHyunJin) [inline external/libtuv/source/tinyara/uv_tinyara_signal.c]: index is 0 or more and less than 'CONFIG_MAX_TASKS'. I wonder why check index.

**Comment by seinfra** [comment]
Target : [e6d77e4b4af3ecaf20965b8ae1c032959e5ea7e1] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e6d77e4b4af3ecaf20965b8ae1c032959e5ea7e1] - Code Rule Check OK.

File: external/libtuv/source/tinyara/uv_tinyara_signal.c (hunk 2 above)
Comment (jeongchanKim) [reply external/libtuv/source/tinyara/uv_tinyara_signal.c]: Yes, you are right. But I think it is good for checking the validation for index.

**Comment by seinfra** [comment]
Target : [90ced36e0f7548c7789dddf0fa912b36141becde] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [90ced36e0f7548c7789dddf0fa912b36141becde] - Code Rule Check OK.

File: external/libtuv/source/tinyara/uv_tinyara_signal.c (hunk 2 above)
Comment (sunghan-chang) [reply external/libtuv/source/tinyara/uv_tinyara_signal.c]: Could you let me know whether it is a valid checking or not?

**Comment by seinfra** [comment]
Target : [987aa11a8f41b69134c3e6b3dd752c828974185e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [987aa11a8f41b69134c3e6b3dd752c828974185e] - Code Rule Check OK.

**Pull request merged**